XiEditor/EditView.swift

```diff
@@ -166,9 +114,15 @@ class EditView: NSView, NSTextInputClient {
         path2.fill()
         */
 
+        let defaultStyle = dataSource.styleMap.defaultStyle
+
+        // set view background
+        self.wantsLayer = defaultStyle.bgColor.alphaComponent != 0.0
+        self.layer?.backgroundColor = defaultStyle.bgColor.cgColor
```

Comment:
In any case, I don't think we should be modifying our layer directly here:

[wantsLayer](https://developer.apple.com/reference/appkit/nsview/1483695-wantslayer#):
>In a layer-backed view, any drawing done by the view is cached to the underlying layer object. This cached content can then be manipulated in ways that are more performant than redrawing the view contents explicitly. AppKit automatically creates the underlying layer object (using the makeBackingLayer() method) and handles the caching of the view’s content. If the wantsUpdateLayer method returns false, you should not interact with the underlying layer object directly. Instead, use the methods of this class to make any changes to the view and its layer. If wantsUpdateLayer returns true, it is acceptable (and appropriate) to modify the layer in the view’s updateLayer() method.

At some point in the future I expect us to go down to the CALayer level for all of our drawing, but for now I think we should stick with the higher level drawing API, and just use `NSRectFill`.

Keep the bg stuff in here, I think; when the style changes fire `editView.needsDisplay = True` to invalidate the view's contents, and then it will use the new defaultStyle on the next drawing cycle.